.travis.yml

```diff
@@ -1,27 +1,44 @@
 # vim: set sw=2 ts=2 softtabstop=2 expandtab:
 language: csharp
-dist: trusty
+dist: bionic
 sudo: true
-solution: "Source/Boogie.sln"
+dotnet: 3.1
+mono: none
+git:
+  depth: false
 env:
   global:
+    - SOLUTION=Source/Boogie-NetCore.sln
     - Z3URL=https://github.com/Z3Prover/z3/releases/download/z3-4.8.4/z3-4.8.4.d6df51951f4c-x64-ubuntu-14.04.zip
   matrix:
-    - BOOGIE_CONFIG=Debug
-    - BOOGIE_CONFIG=Release
+    - CONFIGURATION=Debug
+    - CONFIGURATION=Release
 install:
   # Download a Z3 release
   - wget ${Z3URL}
   - unzip z3*.zip
-  # NuGet is a little flakey in legacy TravisCI, use travis_retry command to retry the command if it fails
-  - travis_retry nuget restore ${TRAVIS_SOLUTION}
-  - travis_retry nuget install -Version 3.10.0 NUnit.Console -OutputDirectory Source/packages/
+  - export PATH="$(find $PWD/z3* -name bin -type d):$PATH"
   # Install needed python tools
   - sudo pip install lit OutputCheck pyyaml
 script:
-  - msbuild /p:Configuration=${BOOGIE_CONFIG} ${TRAVIS_SOLUTION}
-  # Run unit tests
-  - python Source/UnitTests/run-unittests.py ${BOOGIE_CONFIG}
-  # Run driver tests
-  - ln -s $(find $PWD/z3* -name z3 -type f) Binaries/z3.exe
-  - lit -v Test/
+  - dotnet build -c ${CONFIGURATION} ${SOLUTION}
+  - dotnet test --no-build -c ${CONFIGURATION} ${SOLUTION}
+  - lit -v -D dotnet -D configuration=${CONFIGURATION} Test
+deploy:
+  - provider: script
+    script:
+      - dotnet nuget push Source/BoogieDriver/bin/${CONFIGURATION}/Boogie*.nupkg -k ${NUGET_API_KEY} -s https://api.nuget.org/v3/index.json
```

Comment:
Wouldn't the path `Source/BoogieDriver/nupkg/Boogie.2.4.2.nupkg` be better here?

Reply:
Any idea for a clean way to get rid of the wildcard?

Reply:
No, but this seems clean already. Are you worried that this could match multiple files in a single travis build?

Reply:
It's more about knowing that there is a single file and still using a wildcard. But you are right, this is probably more stable than some way of trying to compute the exact file name. It's similar to the code that puts Z3 into PATH. We won't have to change it when we switch to a newer version.